hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/util/S3Utils.java

```diff
@@ -116,4 +118,11 @@ public static S3StorageType toS3StorageType(String storageType)
       throw newError(INVALID_ARGUMENT, storageType, ex);
     }
   }
+
+  public static WebApplicationException wrapOS3Exception(OS3Exception ex) {
+    return new WebApplicationException(ex.getErrorMessage(), ex,
+        Response.status(ex.getHttpCode())
+            .entity(ex.toXml())
+            .build());
+  }
```

Comment:
Nit: `AuthorizationFilter` also seems to use the same method, we can also replace the `wrapOS3Exception` in `AuthorizationFilter` with this.